app/models/chat_channel.rb

```diff
@@ -139,68 +148,29 @@ def title_from_chatable
     end
   end
 
+  private
+
   def change_status(acting_user, target_status)
-    return if !ChatChannel.statuses.include?(target_status.to_s)
     return if !Guardian.new(acting_user).can_change_channel_status?(self, target_status)
-    old_status = self.status
```

Comment:
No need for `old_status`/`new_status` variables as AR provides those (`status` for the current one and `status_previously_was` for the one that was set before saving the model).